feat: new delete message

yqrashawn opened the pull request:
Related: #14188
Corresponding status-go pr https://github.com/status-im/status-go/pull/2922

### Summary

add undo logic to delete similar to delete-for-me

* [x] undo logic
* [x] display deleted messages logic
* [x] deleted messages UI
* [x] reply deleted message UI https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System?node-id=10364%3A123939&t=KxTsDEHlb7o8WlHm-4
* [x] unpin when deleted for real and has the permission to unpin
* [x] don't allow delete for me for pinned messages
* [ ] delete pinned message when don't have perms to unpin the message https://github.com/status-im/status-mobile/issues/14488
* [ ] undo deletion toast UI (will implement this in separate PR) https://github.com/status-im/status-mobile/pull/14376
* [ ] deleted by xxx UI, may need to add `deleted-by` in status-go https://github.com/status-im/status-mobile/issues/14489

### Testing notes

- Deleted message won't disappear, but shown as a system message indicating this is a deleted message.
- Will only broadcast delete action after 4s of delay


status: ready

yqrashawn committed:
feat: delete message undo logic

Signed-off-by: yqrashawn <namy.19@gmail.com>

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 363129b1 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-14232/1/) | 2022-10-26 09:11:31 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-14232/1/consoleText) |

yqrashawn committed:
test: unit test for delete message logic

yqrashawn commented on src/status_im/chat/models/input.cljs:
```diff
@@ -1,20 +1,20 @@
             [goog.object :as object]
             [re-frame.core :as re-frame]
-            [taoensso.timbre :as log]
             [status-im.chat.constants :as chat.constants]
-            [status-im.ethereum.json-rpc :as json-rpc]
             [status-im.chat.models :as chat]
+            [status-im.chat.models.mentions :as mentions]
             [status-im.chat.models.message :as chat.message]
             [status-im.chat.models.message-content :as message-content]
             [status-im.constants :as constants]
+            [status-im.ethereum.json-rpc :as json-rpc]
             [status-im.i18n.i18n :as i18n]
             [status-im.utils.datetime :as datetime]
             [status-im.utils.fx :as fx]
-            ["emojilib" :as emojis]
-            [status-im.chat.models.mentions :as mentions]
-            [status-im.utils.utils :as utils]))
+            [status-im.utils.utils :as utils]
+            [taoensso.timbre :as log]))
```
reorder only

yqrashawn commented on src/status_im/chat/models/input.cljs:
```diff
@@ -133,16 +133,6 @@
              (update-in [:chat/inputs current-chat-id :metadata]
                         dissoc :sending-image))}))
 
-(fx/defn soft-delete-message
-  "Does't delete from db, this is a soft delete"
-  {:events [:chat.ui/soft-delete-message]}
-  [{:keys [db] :as cofx} {:keys [message-id chat-id]}]
-  {::json-rpc/call [{:method      "wakuext_deleteMessageAndSend"
-                     :params      [message-id]
-                     :js-response true
-                     :on-error    #(log/error "failed to delete message message " %)
-                     :on-success  #(re-frame/dispatch [:sanitize-messages-and-process-response %])}]})
-
```
use separate ns

yqrashawn commented on src/status_im/chat/models/message.cljs:
```diff
@@ -183,7 +183,7 @@
 (fx/defn delete-message
   "Deletes chat message, rebuild message-list"
-  {:events [:chat.ui/delete-message]}
+  {:events [:chat.ui/delete-message-not-used-any-more]}
```
This is not used anymore

yqrashawn commented on src/status_im/chat/models/message.cljs:
```diff
@@ -208,10 +208,7 @@
                                                                           [message-id]
                                                                           #(re-frame/dispatch [:chat/decrease-unviewed-count chat-id %3])))) removed-messages)
         remove-messages-fx (fn [{:keys [db]}]
-                             {:db (reduce (fn [acc current]
-                                            (update-in acc [:messages (:chatId current)] dissoc (:messageId current)))
-                                          db removed-messages)
-                              :dispatch-n [[:get-activity-center-notifications]
```
So that deleted messages won't get removed.

yqrashawn commented on src/status_im/ui/screens/chat/sheets.cljs:
```diff
@@ -177,7 +177,7 @@
        :icon                :main-icons/delete
        :accessibility-label :delete-transaccent-button
-       :on-press            #(hide-sheet-and-dispatch [:chat.ui/delete-message chat-id message-id])}]]))
+       :on-press            #(hide-sheet-and-dispatch [:chat.ui/delete-message-not-used-any-more chat-id message-id])}]]))
```
Not in ui2.

yqrashawn commented on src/status_im/ui2/screens/chat/messages/message.cljs:
```diff
@@ -277,17 +278,20 @@
-    (if deleted-for-me?
+    (if (or deleted? deleted-for-me?)
       [system-message/system-message
        {:type             :deleted
-        :label            :message-deleted-for-you
+        :label            (if deleted? :message-deleted :message-deleted-for-you)
         :timestamp-str    timestamp-str
         :non-pressable?   true
-        :animate-landing? (if deleted-for-me-undoable-till true false)}]
+        :animate-landing? (if (or deleted-undoable-till deleted-for-me-undoable-till)
+                            true
+                            false)}]
```
Deleted and deleted-for-me use the same `system-message` type, only difference is the label

ilmotta commented on src/status_im/chat/models/delete_message.cljs:
```diff
@@ -0,0 +1,90 @@
+     ::json-rpc/call [{:method      "wakuext_deleteMessageAndSend"
+                       :params      [message-id]
+                       :js-response true
+                       :on-error    #(log/error "failed to delete message " %)
```
I've recently started to think a bit more about error handling in the Status mobile app.

One of the first things I do is evaluate what's the necessary data that should be logged in order to improve _observability_. In this particular case @yqrashawn, it could be a good idea to log the message ID.

One other point of view is, can we trust the backend is logging all the necessary data in case an RPC call fails? What information is contained in the RPC error object? Is the error object complete enough in case an error occurs? If it's complete enough, we might not need to log anything at all in the frontend. There are tons of stuff to review for error handling, it'd be too boring to discuss that over here. Nonetheless, I'm leaving a comment simply to raise awareness in the team.

yqrashawn replied:
> it could be a good idea to log the message ID
> can we trust the backend is logging all the necessary data in case an RPC call fails? What information is contained in the RPC error object? 

There's definitely no message ID in RPC error message if passing a wrong type of message ID (JSON unmarshal error), so better add message ID here

ilmotta replied:
I see from recent commits you added the message ID to the log in case an error happens, so we're gold now ;)

----

The following is not a suggestion for change, but rather something you might like to use, just as I do. I usually prefer to log errors by passing a human readable string, but also an additional map to contextualize the error in a parseable way. The timbre lib allows (and even encourages) us to this. Of course there are tons of nuances for how a team decides what to log, but you get the idea. For example:

```clojure
#(log/error "Failed to delete message" {:message-id message-id
                                        :error      %})
```

ilmotta commented on src/status_im/chat/models/delete_message_test.cljs:
```diff
@@ -0,0 +1,92 @@
+   [status-im.chat.models.delete-message :as delete-message]
+   [status-im.utils.datetime             :as datetime]))
+
+(defonce mid "message-id")
```
`defonce` is often used for stateful data, is there a particular reason you chose to use `defonce` instead of just `def`?

yqrashawn replied:
Done

ilmotta commented on src/status_im/chat/models/delete_message_test.cljs:
```diff
@@ -0,0 +1,92 @@
+          (is (= (:id result-message) mid))
+          (is (true? (:deleted? result-message)))
+          (is (= (:deleted-undoable-till result-message) 1001))))
+      (testing "should return nil if message not in db"
```
For consistency sake, something most teams agree early on is on how tests descriptions should be written. Given we are in the early days of increasing test coverage, you could try to be consistent in the namespace level.

My own convention is to write something actionable, in the 3rd person, usually starting with a verb. I usually avoid the `should` word because it ends up being superfluous, because one could say that every test _should do something_. If you prefer to use the word should, you could reword tests to always use them, etc, it's up to you of course. In the future, and with more tests, we can standardize a bit more.

Examples translated for your PR:

- `returns nil if message not in db`
- `remains deleted when undo delete late`
- etc

yqrashawn replied:
Done

ilmotta replied:
Thanks!

ilmotta commented on src/status_im/chat/models/delete_message_test.cljs:
```diff
@@ -0,0 +1,92 @@
+                                      :deleted-undoable-till (- (datetime/timestamp) 1000))
+            result-message (get-in (delete-message/undo {:db db} message) [:db :messages cid mid])]
+        (is (= (:id result-message) mid))
+        (is (nil? (:deleted-undoable-till result-message)))
```
Do we really need to assert against `nil?` or `true?`? It's not wrong in any way, but In Clojure, Ruby and other languages, it's very common to take advantage of truthy/falsey values because we don't need to check explicitly for types. For example:

```clojure
(is (not (:deleted-undoable-till result-message)))
(is (:deleted? result-message))
```

yqrashawn replied:
I prefer to assert these values against `nil?` and `true?`, since these values aren't supposed to be set directly by functions in other namespaces.

ilmotta commented on src/status_im/chat/models/delete_message.cljs:
```diff
@@ -0,0 +1,90 @@
+  "Get all deleted messages that not yet synced with status-go and send them"
+  {:events [:chat.ui/send-all-deleted-messages]}
+  [{:keys [db]}]
+  {:dispatch-n (reduce-kv chats-reducer [] (:messages db))})
```
I'm sharing this snippet just in case you like it because I found the reduce-kv with another inner reduce-kv call a bit difficult to understand. When I see a reduce call with a filtering step, I think: _Can we decompose it as a `filter + map` and would that improve readability?_ For the trained Clojure eye, a `filter + map` tells a story without even having to scan the code, whereas the reducer function requires a more thorough scan. Please, forgive me if I got your implementation wrong ;)

```clojure
(fx/defn send-all
  {:events [:chat.ui/send-all-deleted-messages]}
  [{:keys [db]}]
  {:dispatch-n (->> db
                    :messages
                    vals
                    (filter #(and (:deleted? %) (:deleted-undoable-till %)))
                    (mapv #(vector :chat.ui/delete-message-and-send chat-id (:id %))))})
```

yqrashawn replied:
Done

ilmotta reviewed (commented):
I left a bunch of comments that I found valuable, but PR LGTM :rocket:

pavloburykh commented:
@yqrashawn thanx for the PR. could you please clarify if PR is ready for manual testing? 

Currently I don't see any of UI changes that should be introduced by this PR. 

### ISSUE 1 Unable to undo delete action

**Steps:** 

1. Start any chat
2. Send a message
3. Delete a message
4. See if it is possible to undo delete action

**Actual result:** undo element is not displayed. User is unable to undo delete action. 

**Expected result:** element with timer and Undo button is not displayed after message deletion 


https://user-images.githubusercontent.com/97245802/199023332-7e88ade8-a8e9-4ba6-a858-fa118ba2f158.mp4

pavloburykh commented:
@yqrashawn also, could you please rebase PR to the latest develop? Thanx

yqrashawn commented:
@pavloburykh not yet, waiting for the status-go pr https://github.com/status-im/status-go/pull/2922
The status-go changes not included in this PR yet.

pavloburykh commented:
> @pavloburykh not yet, waiting for the status-go pr [status-im/status-go#2922](https://github.com/status-im/status-go/pull/2922) The status-go changes not included in this PR yet.

oh, got, thanx! Please, ping me up when it will be ready for manual testing.

yqrashawn committed:
refactor: changes based on reviews

Signed-off-by: yqrashawn <namy.19@gmail.com>

yqrashawn committed:
fix: sync pending deletes when leaving chatview

yqrashawn commented:
status-go pr merged, icon gone after a big rebase, will look into this tomorrow, convert to draft for now

du64 commented:
What's the point of the "Deleted message" thing in the chat? If I wanted to hide that I deleted a message, having a "Deleted message" notice in chat is the opposite of helpful. Please offer a way to disable this feature.

yqrashawn committed:
fix: system-message icons and labels

yqrashawn commented:
@du64 good point, I'll let the designer know.
@pavloburykh ready for test

yqrashawn committed:
refactor: put data in map for errors

status-im-auto commented:
## 67% of end-end tests have passed
```
Total executed tests: 6
Failed tests: 2
Passed tests: 4
Not executed tests: 1
```
```
IDs of not executed tests: 702777 
```
```
IDs of failed tests: 702733,702745 
```
<h3>Not executed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14232.apk&PR_ID=14232&TR_CASE_IDS=702777">Rerun not executed tests</a></li></details><h3>Failed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14232.apk&PR_ID=14232&TR_CASE_IDS=702733,702745">Rerun failed tests</a></li><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047178">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 1: Getting PN by 'DELETE ME' </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:1352: in test_1_1_chat_text_message_delete_push_disappear
    self.errors.verify_no_errors()
base_test_case.py:188: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Deleted message is shown in chat view for 1-1 chat
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6f73b337c26f4959a3b2fdc1817bbd47?auth=5c601f27314e84b658e579d8e14dc53d#132">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/6f73b337c26f4959a3b2fdc1817bbd47/0037screenshot.png?auth=5c601f27314e84b658e579d8e14dc53d">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cf46782adb847f19f4a502ca1f63b95?auth=aa3213f3753c9b91a8af4014faca3e34#100">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/4cf46782adb847f19f4a502ca1f63b95/0034screenshot.png?auth=aa3213f3753c9b91a8af4014faca3e34">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047179">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><p><blockquote><div>Device 2: Looking for a message by text: hello </div><div>Device 2: Find `TimeStampText` by `xpath`: `//*[starts-with(@text,'hello')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='message-timestamp']` </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:1285: in test_1_1_chat_non_latin_messages_stack_update_profile_photo
    timestamp = self.chat_2.chat_element_by_text(messages[0]).timestamp
../views/chat_view.py:158: in timestamp
    return TimeStampText(self.driver, self.locator).text
../views/base_element.py:204: in text
    return self.find_element().text
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 2: TimeStampText by xpath: `//*[starts-with(@text,'hello')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='message-timestamp']` is not found on the screen
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6f73b337c26f4959a3b2fdc1817bbd47?auth=5c601f27314e84b658e579d8e14dc53d#86">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/6f73b337c26f4959a3b2fdc1817bbd47/0037screenshot.png?auth=5c601f27314e84b658e579d8e14dc53d">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cf46782adb847f19f4a502ca1f63b95?auth=aa3213f3753c9b91a8af4014faca3e34#74">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/4cf46782adb847f19f4a502ca1f63b95/0034screenshot.png?auth=aa3213f3753c9b91a8af4014faca3e34">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (4)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestDeeplinkOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047182">test_public_chat_open_using_deep_link</a>, id: 702776 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5d2df2ae4bb04d6dacd968f15d603d31?auth=9cc206c0df2ebba469e1db7bd576721d#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047180">test_deep_link_with_invalid_user_public_key_own_profile_key</a>, id: 702774 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5d2df2ae4bb04d6dacd968f15d603d31?auth=9cc206c0df2ebba469e1db7bd576721d#40">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047181">test_deep_link_open_user_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5d2df2ae4bb04d6dacd968f15d603d31?auth=9cc206c0df2ebba469e1db7bd576721d#97">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047177">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6f73b337c26f4959a3b2fdc1817bbd47?auth=5c601f27314e84b658e579d8e14dc53d#66">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cf46782adb847f19f4a502ca1f63b95?auth=aa3213f3753c9b91a8af4014faca3e34#53">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

pavloburykh commented:
@yqrashawn thanx for the PR. Please, take a look at the following issues

### ISSUE 1 Message is not displayed as deleted for the receiver if his chat screen was opened during message has been deleted by sender

**Steps:**
1. Start any chat between User A and User B
2. Keep chat screen opened by both users
3. Send some message by User A
4. Delete for everyone message by User A
5. See if message has been deleted for User B

**Actual result:** message remains undeleted for User B until he re-opens the chat
**Expected result:** message is displayed as deleted for User B without chat re-opening


https://user-images.githubusercontent.com/97245802/202691572-b8658497-d1c2-40ff-b1b8-1b5ceb9a4190.mp4

[sender_geth.log](https://github.com/status-im/status-mobile/files/10040746/sender_geth.log)
[sender_Status.log](https://github.com/status-im/status-mobile/files/10040747/sender_Status.log)
[receiver_geth.log](https://github.com/status-im/status-mobile/files/10040749/receiver_geth.log)
[receiver_Status.log](https://github.com/status-im/status-mobile/files/10040750/receiver_Status.log)

pavloburykh commented:
### ISSUE 2 Reactions for the deleted message are not deleted together with the message

**Steps:**
1. Start any chat between User A and User B
3. Send some message by User A
4. Set reaction to the message (by any user)
5. Delete for everyone message by User A
6. See if reactions are also deleted together with the message

**Actual result:** reactions are deleted
**Expected result:** reactions element should be deleted together with the message


https://user-images.githubusercontent.com/97245802/202693437-dfa5973d-1b00-4288-943c-a4c96099e9ea.mp4

pavloburykh commented:
@yqrashawn could you also please resolve conflicts in src/status_im/utils/config.cljs ?

yqrashawn commented:
@pavloburykh rebased and able to reproduce both issues on my side

pavloburykh commented:
> @pavloburykh rebased and able to reproduce both issues on my side I'll ping you once it's fixed

Thank you @yqrashawn! When PR will be ready for testing, please move it to the E2E column on the board.

yqrashawn committed:
fix: handle received removed-messages

yqrashawn committed:
fix: don't render reactions for deleted messages

Signed-off-by: yqrashawn <namy.19@gmail.com>

yqrashawn commented:
@pavloburykh ready for test

status-im-auto commented:
## 83% of end-end tests have passed
```
Total executed tests: 6
Failed tests: 1
Passed tests: 5
Not executed tests: 1
```
```
IDs of not executed tests: 702777 
```
```
IDs of failed tests: 702745 
```
<h3>Not executed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14232.apk&PR_ID=14232&TR_CASE_IDS=702777">Rerun not executed tests</a></li></details><h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14232.apk&PR_ID=14232&TR_CASE_IDS=702745">Rerun failed tests</a></li><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047267">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><p><blockquote><div>Device 2: Looking for a message by text: hello </div><div>Device 2: Find `TimeStampText` by `xpath`: `//*[starts-with(@text,'hello')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='message-timestamp']` </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:1285: in test_1_1_chat_non_latin_messages_stack_update_profile_photo
    timestamp = self.chat_2.chat_element_by_text(messages[0]).timestamp
../views/chat_view.py:160: in timestamp
    return TimeStampText(self.driver, self.locator).text
../views/base_element.py:205: in text
    return self.find_element().text
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 2: TimeStampText by xpath: `//*[starts-with(@text,'hello')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='message-timestamp']` is not found on the screen
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3bffe5bd5ab14d6db74e61fa68389592?auth=03aa0f83cd5465d2bb003dc1c7f13265#86">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/3bffe5bd5ab14d6db74e61fa68389592/0037screenshot.png?auth=03aa0f83cd5465d2bb003dc1c7f13265">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b356160efd9c402aa1f9bd696f3e78a8?auth=2b93150cf54fbe6725b4fd827ce6e573#74">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/b356160efd9c402aa1f9bd696f3e78a8/0034screenshot.png?auth=2b93150cf54fbe6725b4fd827ce6e573">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (5)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047266">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3bffe5bd5ab14d6db74e61fa68389592?auth=03aa0f83cd5465d2bb003dc1c7f13265#132">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b356160efd9c402aa1f9bd696f3e78a8?auth=2b93150cf54fbe6725b4fd827ce6e573#100">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047265">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3bffe5bd5ab14d6db74e61fa68389592?auth=03aa0f83cd5465d2bb003dc1c7f13265#66">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b356160efd9c402aa1f9bd696f3e78a8?auth=2b93150cf54fbe6725b4fd827ce6e573#53">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeeplinkOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047270">test_public_chat_open_using_deep_link</a>, id: 702776 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2bbe4437d9724d4c9f64cd29bd09e19a?auth=f2a02b0ec9d67cc477814c0a8452c748#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047268">test_deep_link_with_invalid_user_public_key_own_profile_key</a>, id: 702774 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2bbe4437d9724d4c9f64cd29bd09e19a?auth=f2a02b0ec9d67cc477814c0a8452c748#40">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047269">test_deep_link_open_user_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2bbe4437d9724d4c9f64cd29bd09e19a?auth=f2a02b0ec9d67cc477814c0a8452c748#97">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

pavloburykh commented:
@yqrashawn thanx for your fixes. 

[ISSUE 2](https://github.com/status-im/status-mobile/pull/14232#issuecomment-1319869102) is partially reproducible: reactions are still not deleted if message is Deleted for me. 


https://user-images.githubusercontent.com/97245802/203079908-1cb4884f-a44e-4d4b-ba8a-7f8fbd32fcaa.mp4

yqrashawn commented:
> reactions are still not deleted if message is Deleted for me.

Oops, sorry for that, fixed with new push

status-im-auto commented:
## 83% of end-end tests have passed
```
Total executed tests: 6
Failed tests: 1
Passed tests: 5
Not executed tests: 1
```
```
IDs of not executed tests: 702777 
```
```
IDs of failed tests: 702745 
```
<h3>Not executed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14232.apk&PR_ID=14232&TR_CASE_IDS=702777">Rerun not executed tests</a></li></details><h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14232.apk&PR_ID=14232&TR_CASE_IDS=702745">Rerun failed tests</a></li><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047281">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><p><blockquote><div>Device 2: Looking for a message by text: hello </div><div>Device 2: Find `TimeStampText` by `xpath`: `//*[starts-with(@text,'hello')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='message-timestamp']` </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:1285: in test_1_1_chat_non_latin_messages_stack_update_profile_photo
    timestamp = self.chat_2.chat_element_by_text(messages[0]).timestamp
../views/chat_view.py:160: in timestamp
    return TimeStampText(self.driver, self.locator).text
../views/base_element.py:205: in text
    return self.find_element().text
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 2: TimeStampText by xpath: `//*[starts-with(@text,'hello')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='message-timestamp']` is not found on the screen
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d9ea2e57549a4ab28f33b42aef51f304?auth=32afd124b2021203a1c59385aa580a49#86">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/d9ea2e57549a4ab28f33b42aef51f304/0037screenshot.png?auth=32afd124b2021203a1c59385aa580a49">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/db4bf438916a4d41af282c573e8bf309?auth=be062e5fc64ddc7f5bc8e064249c5345#74">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/db4bf438916a4d41af282c573e8bf309/0034screenshot.png?auth=be062e5fc64ddc7f5bc8e064249c5345">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (5)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047280">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d9ea2e57549a4ab28f33b42aef51f304?auth=32afd124b2021203a1c59385aa580a49#132">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/db4bf438916a4d41af282c573e8bf309?auth=be062e5fc64ddc7f5bc8e064249c5345#100">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047279">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d9ea2e57549a4ab28f33b42aef51f304?auth=32afd124b2021203a1c59385aa580a49#66">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/db4bf438916a4d41af282c573e8bf309?auth=be062e5fc64ddc7f5bc8e064249c5345#53">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeeplinkOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047284">test_public_chat_open_using_deep_link</a>, id: 702776 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cdaef39d16c24455ba43b56963c69d64?auth=4bc5adf874ba9d07db296c729803056d#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047282">test_deep_link_with_invalid_user_public_key_own_profile_key</a>, id: 702774 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cdaef39d16c24455ba43b56963c69d64?auth=4bc5adf874ba9d07db296c729803056d#40">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3047283">test_deep_link_open_user_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cdaef39d16c24455ba43b56963c69d64?auth=4bc5adf874ba9d07db296c729803056d#97">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

pavloburykh commented:
@yqrashawn thank you. I have found another issue, please take a look

### ISSUE 3 The last messages are not deleted if there are more than 1 deleted messages in a row

**Steps:**

1. Send more than 1 message (better 5) in any chat
2. Start deleting messages one after another
3. Re-open the chat
4. See if all messages are preserved as deleted

**Actual result:** the last 1 or 2 messages that have been deleted are displayed as undeleted

**Expected result:** all messages should remain deleted


https://user-images.githubusercontent.com/97245802/203297621-2340b76e-43ae-4c37-b1cf-e0729970ea81.mp4

[geth.log](https://github.com/status-im/status-mobile/files/10066704/geth.log)
[Status.log](https://github.com/status-im/status-mobile/files/10066706/Status.log)

yqrashawn committed:
fix: sync/send all pending deletes when leave chat

Signed-off-by: yqrashawn <namy.19@gmail.com>

status-im-auto commented:
## 75% of end-end tests have passed
```
Total executed tests: 4
Failed tests: 1
Passed tests: 3
Not executed tests: 3
```
```
IDs of not executed tests: 702730,702731,702745 
```
```
IDs of failed tests: 702733 
```
<h3>Not executed tests (3)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14232.apk&PR_ID=14232&TR_CASE_IDS=702730,702731,702745">Rerun not executed tests</a></li></details><h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14232.apk&PR_ID=14232&TR_CASE_IDS=702733">Rerun failed tests</a></li><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3048445">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:1225: in prepare_devices
    self.drivers, self.loop = create_shared_drivers(2)
base_test_case.py:306: in create_shared_drivers
    drivers = loop.run_until_complete(start_threads(quantity,
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:21: in start_threads
    returns[k] = await returns[k]
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
/home/jenkins/.local/lib/python3.10/site-packages/appium/webdriver/webdriver.py:151: in __init__
    super().__init__(
/home/jenkins/.local/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:157: in __init__
    self.start_session(capabilities, browser_profile)
/home/jenkins/.local/lib/python3.10/site-packages/appium/webdriver/webdriver.py:225: in start_session
    response = self.execute(RemoteCommand.NEW_SESSION, parameters)
/home/jenkins/.local/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:319: in execute
    response = self.command_executor.execute(driver_command, params)
/home/jenkins/.local/lib/python3.10/site-packages/selenium/webdriver/remote/remote_connection.py:376: in execute
    return self._request(command_info[0], url, body=data)
/home/jenkins/.local/lib/python3.10/site-packages/selenium/webdriver/remote/remote_connection.py:399: in _request
    resp = self._conn.request(method, url, body=body, headers=headers)
/home/jenkins/.local/lib/python3.10/site-packages/urllib3/request.py:78: in request
    return self.request_encode_body(
/home/jenkins/.local/lib/python3.10/site-packages/urllib3/request.py:170: in request_encode_body
    return self.urlopen(method, url, **extra_kw)
/home/jenkins/.local/lib/python3.10/site-packages/urllib3/poolmanager.py:418: in urlopen
    return self.urlopen(method, redirect_location, **kw)
/home/jenkins/.local/lib/python3.10/site-packages/urllib3/poolmanager.py:418: in urlopen
    return self.urlopen(method, redirect_location, **kw)
/home/jenkins/.local/lib/python3.10/site-packages/urllib3/poolmanager.py:418: in urlopen
    return self.urlopen(method, redirect_location, **kw)
/home/jenkins/.local/lib/python3.10/site-packages/urllib3/poolmanager.py:405: in urlopen
    retries = retries.increment(method, url, response=response, _pool=conn)
/home/jenkins/.local/lib/python3.10/site-packages/urllib3/util/retry.py:592: in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
 HTTPSConnectionPool(host='ondemand.eu-central-1.saucelabs.com', port=443): Max retries exceeded with url: https://anton-status:0a8c5013-d483-4872-a894-9b62d6391892@ondemand.eu-central-1.saucelabs.com:443/wd/hub/session/cbd13078756f434b88dd10114ef97ffc?ref=b35f84a4143a4bff9478101ec918caab (Caused by ResponseError('too many redirects'))
```

<br/><br/></td></tr></tbody></table></details><h3>Passed tests (3)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestDeeplinkOneDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3048449">test_public_chat_open_using_deep_link</a>, id: 702776 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/65d87881c3c642bba1aed18fbfa365ef?auth=7228bfe9a1b2da482275825630054a5d#80">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3048447">test_deep_link_with_invalid_user_public_key_own_profile_key</a>, id: 702774 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/65d87881c3c642bba1aed18fbfa365ef?auth=7228bfe9a1b2da482275825630054a5d#41">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3048448">test_deep_link_open_user_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/65d87881c3c642bba1aed18fbfa365ef?auth=7228bfe9a1b2da482275825630054a5d#98">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

pavloburykh commented:
@yqrashawn PR has been automatically moved to TO REBASE column, because of existing conflicts. Could you please resolve conflicts and then move PR back to E2E column when it will be ready for testing. 

You can also find information regarding current PR workflow process here https://notes.status.im/M73LlV1gT3OJ8kdSMY-ZQQ?view

Thanx your work and patience!

yqrashawn commented:
@pavloburykh thanks, ready for test

yqrashawn commented:
rebased again

VladimrLitvinenko commented:
Hi @yqrashawn. Thank you for your PR. Take a look the following issues:

**ISSUE 1 'Deleted for everyone' pinned message is not removed from list of pinned messages**
**Steps to reproduce:**
1. Create 1-1 chat as User_A with User_B
2. Create message within chat
3. Pin this message
4.  Remove this message using 'Deleted for everyone' option

**Actual result:**
'Deleted for everyone' message is shown within list of pinned messages
https://user-images.githubusercontent.com/52490791/204040386-de71ccb5-1694-4d06-8c55-82f600979343.mp4


**Expected result:**
The 'Delete for everyone' message is removed from the list of pinned messages 

**ISSUE 2. 'Deleted for me' message is removed for both users within list of pinned message** 
**Steps to reproduce:**
1. Create 1-1 chat as User_A with User_B
2. Create message within chat
3. Pin this message
4.  Remove this message using 'Deleted for me' option

**Actual result:**
'Deleted for me' message is removed for both users
https://user-images.githubusercontent.com/52490791/204041475-7d5f989d-a91d-4ef5-8b2b-85abd1dba441.mp4


**Expected result:**
'Deleted for me' message is removed for only user who removed current message from the list of pinned messages

Hi @pedro-et take a look at the expected result of ISSUE 1 is it suitable behavior? I have investigated how it works within Telegram and WhatsApp services and the behavior is the following: if user removes the message as  'Delete for everyone, then, the current message disappears from the list of pinned for all users. Should it disappear in Status or be shown as removed?
![image](https://user-images.githubusercontent.com/52490791/204040138-09f0bd82-b89d-4419-83f5-a0f7bc819577.png)
Also take a look at ISSUE 2 in the expected result section. Is correct behavior that message should be removed from the list of pinned only for user who removes current message?

VladimrLitvinenko commented:
> Hi @yqrashawn. Thank you for your PR. Take a look the following issues. And let's wait on a reply from @pedro-et side
> 
> **ISSUE 4 'Deleted for everyone' pinned message is not removed from list of pinned messages and can't be unpinned** **Steps to reproduce:**
> 
> 1. Create 1-1 chat as User_A with User_B
> 2. Create a message within the chat
> 3. Pin this message
> 4. Remove this message using 'Deleted for everyone' option
> 5. Reopen current chat
> 6. Go to the list of pinned messages
> 7. Try to unpin the removed message
> 
> **Actual result:**
> 
> * 'Deleted for everyone' message is shown within list of pinned messages
> * Removed message can't be unpinned from the list of pinned messages
> 
>  cantunpin.mp4 
> **Expected result:**
> 
> * The 'Delete for everyone' message is removed from the list of pinned messages
> * Removed message is able to be unpinned from the list of pinned messages
> 
> **ISSUE 5. 'Deleted for everyone' message is shown as deleted in the pinned list only after chat is reopened**
> 
> **Steps to reproduce:**
> 
> 1. Create 1-1 chat as User_A with User_B
> 2. Create a message within the chat
> 3. Pin this message
> 4. Remove this message using 'Deleted for everyone' option
> 5. Reopen the current chat
> 
> **Actual result:** 'Deleted for everyone' message is shown with the initial content unless the chat is reopened
> 
>  deleted.mp4 
> **Expected result:** 'Deleted for everyone' message is shown as deleted
> 
> **ISSUE 6. 'Deleted for me' message is removed for both users within list of pinned message** **Steps to reproduce:**
> 
> 1. Create 1-1 chat as User_A with User_B
> 2. Create message within chat
> 3. Pin this message
> 4. Remove this message using 'Deleted for me' option
> 
> **Actual result:** 'Deleted for me' message is removed for both users https://user-images.githubusercontent.com/52490791/204041475-7d5f989d-a91d-4ef5-8b2b-85abd1dba441.mp4
> 
> **Expected result:** 'Deleted for me' message is removed for only user who removed current message from the list of pinned messages
> 
> Hi @pedro-et take a look at the expected result of ISSUE 1 is it suitable behavior? I have investigated how it works within Telegram and WhatsApp services and the behavior is the following: if user removes the message as 'Delete for everyone, then, the current message disappears from the list of pinned for all users. Should it disappear in Status or be shown as removed?
> 
> Currently, it disappears from the list of pinned messages if 'delete for me' option is used, but shown as deleted if 'delete for everyone' option is used and then can't be unpinned ![image](https://user-images.githubusercontent.com/52490791/204040138-09f0bd82-b89d-4419-83f5-a0f7bc819577.png) Also, take a look at ISSUE 2 in the expected result section. Is it correct behavior that the message should be removed from the list of pinned only for the user who removes the current message?

hi @pedro-et 
You may have missed this message over the weekend. Just a mention for you :)

VladimrLitvinenko commented:
Hi @yqrashawn should the showing of the deleted message be in the scope of the current PR?
**On PR:**
![image](https://user-images.githubusercontent.com/52490791/204336131-edf009a2-ae53-419a-9c2d-04bd9092429c.png)

**On design:**
![image](https://user-images.githubusercontent.com/52490791/204335685-93437978-0727-4a37-a01f-cea38f69b288.png)

yqrashawn commented:
> should the showing of the deleted message be in the scope of the current PR?

Yes, any already implemented related features should be covered. 
Do you have some test cases can share with me? @VladimrLitvinenko 

Also PR not ready for test yet. Just a rebase. I'll move this to e2e test once ready for test.

VladimrLitvinenko commented:
> > should the showing of the deleted message be in the scope of the current PR?
> 
> Yes, any already implemented related features should be covered. Do you have some test cases can share with me? @VladimrLitvinenko
> 
> Also PR not ready for test yet. Just a rebase. I'll move this to e2e test once ready for test.

sure. Here is a list of test cases. Let me know if you need some more details https://docs.google.com/document/d/1oi-QarOL86JdGGdFjnB-jFuPby6B2bDGyctZfUtt6Yg/edit?usp=sharing

VladimrLitvinenko commented:
> the



> Hi @yqrashawn. Thank you for your PR. Take a look the following issues. And let's wait on a reply from @pedro-et side
> 
> **ISSUE 4 'Deleted for everyone' pinned message is not removed from list of pinned messages and can't be unpinned** **Steps to reproduce:**
> 
> 1. Create 1-1 chat as User_A with User_B
> 2. Create a message within the chat
> 3. Pin this message
> 4. Remove this message using 'Deleted for everyone' option
> 5. Reopen current chat
> 6. Go to the list of pinned messages
> 7. Try to unpin the removed message
> 
> **Actual result:**
> 
> * 'Deleted for everyone' message is shown within list of pinned messages
> * Removed message can't be unpinned from the list of pinned messages
> 
>  cantunpin.mp4 
> **Expected result:**
> 
> * The 'Delete for everyone' message is removed from the list of pinned messages
> * Removed message is able to be unpinned from the list of pinned messages
> 
> **ISSUE 5. 'Deleted for everyone' message is shown as deleted in the pinned list only after chat is reopened**
> 
> **Steps to reproduce:**
> 
> 1. Create 1-1 chat as User_A with User_B
> 2. Create a message within the chat
> 3. Pin this message
> 4. Remove this message using 'Deleted for everyone' option
> 5. Reopen the current chat
> 
> **Actual result:** 'Deleted for everyone' message is shown with the initial content unless the chat is reopened
> 
>  deleted.mp4 
> **Expected result:** 'Deleted for everyone' message is shown as deleted
> 
> **ISSUE 6. 'Deleted for me' message is removed for both users within list of pinned message** **Steps to reproduce:**
> 
> 1. Create 1-1 chat as User_A with User_B
> 2. Create message within chat
> 3. Pin this message
> 4. Remove this message using 'Deleted for me' option
> 
> **Actual result:** 'Deleted for me' message is removed for both users https://user-images.githubusercontent.com/52490791/204041475-7d5f989d-a91d-4ef5-8b2b-85abd1dba441.mp4
> 
> **Expected result:** 'Deleted for me' message is removed for only user who removed current message from the list of pinned messages
> 
> Hi @pedro-et take a look at the expected result of ISSUE 1 is it suitable behavior? I have investigated how it works within Telegram and WhatsApp services and the behavior is the following: if user removes the message as 'Delete for everyone, then, the current message disappears from the list of pinned for all users. Should it disappear in Status or be shown as removed?
> 
> Currently, it disappears from the list of pinned messages if 'delete for me' option is used, but shown as deleted if 'delete for everyone' option is used and then can't be unpinned ![image](https://user-images.githubusercontent.com/52490791/204040138-09f0bd82-b89d-4419-83f5-a0f7bc819577.png) Also, take a look at ISSUE 2 in the expected result section. Is it correct behavior that the message should be removed from the list of pinned only for the user who removes the current message?

Hi @xAlisher take a look please at the question addressed to Pedro. Pedro is not available, maybe you can help us

yqrashawn commented:
> Re. your questions:
> 1. yes, we allow user to delete a pinned message without unpinning first
> 2. yes, when a pinned message is deleted, at the same time it is deleted it should also be unpinned

John replied in mobile-ui channel @VladimrLitvinenko

VladimrLitvinenko commented:
> > Re. your questions:
> > 
> > 1. yes, we allow user to delete a pinned message without unpinning first
> > 2. yes, when a pinned message is deleted, at the same time it is deleted it should also be unpinned
> 
> John replied in mobile-ui channel @VladimrLitvinenko

Oh. I see, thank you

VladimrLitvinenko commented:
Hi @yqrashawn. fix please additional issues:
**ISSUE 7. The '3 random names'/ ens / Nickname is not shown of user owner who deleted the message using Deleted for everyone option**
**Steps to reproduce:**
1. Open 1-1 chat as USER_A and USER_B
2. Send some message as USER_A
3. Remove the message using 'Delete for everyone'

**Actual result:**
USER_B sees
![image](https://user-images.githubusercontent.com/52490791/204790363-07ec60a8-4748-4536-a130-6502f484ea66.png)

**Expected result:**
USER_B sees
![image](https://user-images.githubusercontent.com/52490791/204789105-bf75d5fd-7f0a-4153-a9ba-ff19f796c6fa.png)
USER_A sees 
![image](https://user-images.githubusercontent.com/52490791/204790565-bb2f7519-6810-4499-9e73-ed86b0882d27.png)

**ISSUE 8. 'Message deleted for everyone' should be shown instead of 'Message deleted'**
**Steps to reproduce:**
1. Open 1-1 chat 
2. Send some message 
3. Remove the message using 'Delete for everyone'
**Actual result:**
![image](https://user-images.githubusercontent.com/52490791/204791501-2a31742d-5a1e-4fc4-b60b-5772a758dda1.png)

**Expected result:**
![image](https://user-images.githubusercontent.com/52490791/204791255-9c25151f-4f4a-4790-b39d-c9a7b0e6d443.png)

***ISSUE 9. The replied message should be shown as deleted**
**Steps to reproduce:**
1. Open 1-1 chat 
2. Send some message 
3. Reply on the message
4. Remove message that was sent in step 2

**Actual result:**
The message is not shown as removed
![image](https://user-images.githubusercontent.com/52490791/204793085-2f0299ad-ff47-4879-85db-e6f1dcc75a48.png)

**Expected result:**
The message is shown as removed
![image](https://user-images.githubusercontent.com/52490791/204791941-049466da-1410-4052-99b8-893a30134f86.png)

yqrashawn commented:
> > > should the showing of the deleted message be in the scope of the current PR?
> > 
> > 
> > Yes, any already implemented related features should be covered. Do you have some test cases can share with me? @VladimrLitvinenko
> > Also PR not ready for test yet. Just a rebase. I'll move this to e2e test once ready for test.
> 
> sure. Here is a list of test cases. Let me know if you need some more details [docs.google.com/document/d/1oi-QarOL86JdGGdFjnB-jFuPby6B2bDGyctZfUtt6Yg/edit?usp=sharing](https://docs.google.com/document/d/1oi-QarOL86JdGGdFjnB-jFuPby6B2bDGyctZfUtt6Yg/edit?usp=sharing)

🤩 Thanks for the test cases.

yqrashawn committed:
feat: add reply deleted message

https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System?node-id=10364%3A123939&t=KxTsDEHlb7o8WlHm-4

VladimrLitvinenko commented:
HI @yqrashawn some updates from John's side, share our discussion from UI-mobile
- hi John_  thank you for answers.One more question. Should the message be unpinned for all users if 'delete for you' option is used? or only unpinned for current user?

VladimrLitvinenko commented:
Hi @yqrashawn updates from John's side that I wanted to share with you. I think this is not in the scope of this PR as UI is not ready in Figma yet
-  **Me:** John_ One more question. Should the message be unpinned for all users  if 'delete for you' option is used? or only unpinned for the current user?
- **John:** Good question! I think we might actually want to now display the 'delete for you' option for any message that's pinned, otherwise is a user who is for example the Owner of a community uses the 'delete for you' option for a pinned message, and the community doesn't have any other admins and if the 'only owners and admins can pin and unpin messages' setting is enabled, then there would be no easy way to unpin the message in the future.
So if we remove the 'delete for you' option from any message that is pinned we can avoid these nasty corner cases.  cc'ing Alisher

yqrashawn committed:
feat: send unpin rpc when send delete to status-go

1. send unpin request when send delete to status-go
2. hide deleted pinned messages

Signed-off-by: yqrashawn <namy.19@gmail.com>

yqrashawn commented:
> Hi @yqrashawn updates from John's side that I wanted to share with you. I think this is not in the scope of this PR as UI is not ready in Figma yet
> 
>     * **Me:** John_ One more question. Should the message be unpinned for all users  if 'delete for you' option is used? or only unpinned for the current user?
> 
>     * **John:** Good question! I think we might actually want to now display the 'delete for you' option for any message that's pinned, otherwise is a user who is for example the Owner of a community uses the 'delete for you' option for a pinned message, and the community doesn't have any other admins and if the 'only owners and admins can pin and unpin messages' setting is enabled, then there would be no easy way to unpin the message in the future.
>       So if we remove the 'delete for you' option from any message that is pinned we can avoid these nasty corner cases.  cc'ing Alisher

Thanks. I just pushed new commit to hide pinned deleted for me/everyone messages. Although not ready for test yet.

yqrashawn committed:
feat: no delete for me option for pinned message

yqrashawn committed:
fix: add message deleted for everyone copy

yqrashawn commented:
@VladimrLitvinenko ready for test 

issues that will be resolved with future PRs

**ISSUE 4 'Deleted for everyone' pinned message is not removed from list of pinned messages and can't be unpinned**
partially fixed, will now try to unpin the message (without checking if user has the permission to unpin message) when syncing deleted pinned message with status-go

**ISSUE 7. The '3 random names'/ ens / Nickname is not shown of user owner who deleted the message using Deleted for everyone option**
not implemented, might require persist “deleted by” info in status-go

I've created follow-up issue for them 
https://github.com/status-im/status-mobile/issues/14488
https://github.com/status-im/status-mobile/issues/14489

status-im-auto commented:
## 100% of end-end tests have passed
```
Total executed tests: 7
Failed tests: 0
Passed tests: 7
```
<h3>Passed tests (7)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3064414">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8d24a8e1a7dd487c8ca8fd5aaef907c4?auth=c9de2ef3cb9dbc6f0089965b5926bc6f#254">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6559c4472f6a4524bd7218a8578bf848?auth=e4cd0ed22c20ce056eef6ea50a36e065#171">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3064415">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8d24a8e1a7dd487c8ca8fd5aaef907c4?auth=c9de2ef3cb9dbc6f0089965b5926bc6f#175">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6559c4472f6a4524bd7218a8578bf848?auth=e4cd0ed22c20ce056eef6ea50a36e065#104">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3064412">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8d24a8e1a7dd487c8ca8fd5aaef907c4?auth=c9de2ef3cb9dbc6f0089965b5926bc6f#66">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6559c4472f6a4524bd7218a8578bf848?auth=e4cd0ed22c20ce056eef6ea50a36e065#53">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3064413">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8d24a8e1a7dd487c8ca8fd5aaef907c4?auth=c9de2ef3cb9dbc6f0089965b5926bc6f#86">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6559c4472f6a4524bd7218a8578bf848?auth=e4cd0ed22c20ce056eef6ea50a36e065#74">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeeplinkOneDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3064418">test_public_chat_open_using_deep_link</a>, id: 702776 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/77bc4f7d0e1d440d95aacf19f4cebe6c?auth=3e95f588fa1322f713f6f9b714df7b94#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3064416">test_deep_link_with_invalid_user_public_key_own_profile_key</a>, id: 702774 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/77bc4f7d0e1d440d95aacf19f4cebe6c?auth=3e95f588fa1322f713f6f9b714df7b94#40">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3064417">test_deep_link_open_user_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/77bc4f7d0e1d440d95aacf19f4cebe6c?auth=3e95f588fa1322f713f6f9b714df7b94#97">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

VladimrLitvinenko commented:
Hi @yqrashawn . Thank you! Test results for current PR will be given soon

VladimrLitvinenko commented:
Hi @yqrashawn . Some issues were found

**ISSUE 10. The pinned message should not be able to be with 'delete for you' option**

**Steps to reproduce:**
1. Create 1-1 chat
2. Send the message as USER_A
3. Pin the message as USER_A
4. Tap on the message as USER_B

**Actual result:**
'Deleted for me' option is still shown
![image](https://user-images.githubusercontent.com/52490791/205676121-41a7de61-c27d-46ce-8b1d-64dc3196c795.png)

**Expected result:**
The 'delete for me option' is not shown for all users If message is pinned

**ISSUE 11. 'Deleted for everyone' option disappears from the chat if to reopen it**

**Steps to reproduce:**
1. Create 1-1 chat
2. Send the message
3. Remove this message using 'Deleted for everyone' option
4. Reopen the chat

**Actual result:**
'Delete for everyone' option is not shown in the chat

https://user-images.githubusercontent.com/52490791/205680865-0f71a8bf-95b1-4cdf-91b4-8a8381050d11.mp4


**Expected result:**
'Delete for everyone' option is shown 


**ISSUE 12. 'Replied message is not shown as deleted for USER_B**

**Steps to reproduce:**
1. Create message in 1-1 chat
2. Send the message as USER_A with text "Hello world 1"
3. Leave reply for current message as USER_A
4. Remove message from step2 as USER_A using 'delete for everyone option'
5. Check the message as USER_B

**Actual result:**
USER_B sees
![image](https://user-images.githubusercontent.com/52490791/205688291-0b96a186-b30a-4f11-b019-73c21bcc0b87.png)

**Expected result:**
USER_B sees that the replied message is deleted
![image](https://user-images.githubusercontent.com/52490791/205688870-b9e7472b-7f56-4eac-ab11-174c17b338a3.png)

yqrashawn committed:
fix: upgrade status-go, fix missing deleted message issue

https://github.com/status-im/status-go/compare/9227c631...a97b5527

yqrashawn commented:
@VladimrLitvinenko issue 11 and 12 are fixed, ready for test again

yqrashawn commented:
* [ ] delete pinned message when don't have perms to unpin the message https://github.com/status-im/status-mobile/issues/14488
* [ ] undo deletion toast UI (will implement this in separate PR) https://github.com/status-im/status-mobile/pull/14376
* [ ] deleted by xxx UI, may need to add `deleted-by` in status-go https://github.com/status-im/status-mobile/issues/14489
* [ ] check code in this PR, see if anything should be move to status-im2
